src/vulkan-renderer/imgui/imgui.cpp

```diff
@@ -0,0 +1,616 @@
+    assert(device.device());
+    assert(graphics_card);
+    assert(vma_allocator);
+    assert(data_transfer_queue);
+
+    spdlog::debug("Creating ImGUI context");
+
+    ImGui::CreateContext();
+
+    ImGuiStyle &style = ImGui::GetStyle();
+
+    spdlog::debug("Setting ImGUI style");
+
+    style.Colors[ImGuiCol_TitleBg] = ImVec4(1.0f, 0.0f, 0.0f, 1.0f);
+    style.Colors[ImGuiCol_TitleBgActive] = ImVec4(1.0f, 0.0f, 0.0f, 1.0f);
+    style.Colors[ImGuiCol_TitleBgCollapsed] = ImVec4(1.0f, 0.0f, 0.0f, 0.1f);
+    style.Colors[ImGuiCol_MenuBarBg] = ImVec4(1.0f, 0.0f, 0.0f, 0.4f);
+    style.Colors[ImGuiCol_Header] = ImVec4(0.8f, 0.0f, 0.0f, 0.4f);
+    style.Colors[ImGuiCol_HeaderActive] = ImVec4(1.0f, 0.0f, 0.0f, 0.4f);
+    style.Colors[ImGuiCol_HeaderHovered] = ImVec4(1.0f, 0.0f, 0.0f, 0.4f);
+    style.Colors[ImGuiCol_FrameBg] = ImVec4(0.0f, 0.0f, 0.0f, 0.8f);
+    style.Colors[ImGuiCol_CheckMark] = ImVec4(1.0f, 0.0f, 0.0f, 0.8f);
+    style.Colors[ImGuiCol_SliderGrab] = ImVec4(1.0f, 0.0f, 0.0f, 0.4f);
+    style.Colors[ImGuiCol_SliderGrabActive] = ImVec4(1.0f, 0.0f, 0.0f, 0.8f);
+    style.Colors[ImGuiCol_FrameBgHovered] = ImVec4(1.0f, 1.0f, 1.0f, 0.1f);
+    style.Colors[ImGuiCol_FrameBgActive] = ImVec4(1.0f, 1.0f, 1.0f, 0.2f);
+    style.Colors[ImGuiCol_Button] = ImVec4(1.0f, 0.0f, 0.0f, 0.4f);
+    style.Colors[ImGuiCol_ButtonHovered] = ImVec4(1.0f, 0.0f, 0.0f, 0.6f);
+    style.Colors[ImGuiCol_ButtonActive] = ImVec4(1.0f, 0.0f, 0.0f, 0.8f);
+
+    ImGuiIO &io = ImGui::GetIO();
+
+    spdlog::debug("Setting ImGUI scale");
+
+    io.FontGlobalScale = m_scale;
+
+    spdlog::debug("Loading ImGUI vertex shader");
+
+    m_vert_shader = std::make_unique<wrapper::Shader>(m_device, VK_SHADER_STAGE_VERTEX_BIT, "ImGUI vertex shader",
+                                                      "shaders/ui.vert.spv");
+
+    VkPipelineShaderStageCreateInfo shader_info;
+
+    shader_info.sType = VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO;
+    shader_info.pNext = nullptr;
+    shader_info.flags = 0;
+    shader_info.stage = VK_SHADER_STAGE_VERTEX_BIT;
+    shader_info.module = m_vert_shader->module();
+    shader_info.pName = m_vert_shader->entry_point().c_str();
+    shader_info.pSpecializationInfo = nullptr;
+
+    shaders.push_back(shader_info);
+
+    spdlog::debug("Loading ImGUI fragment shader");
+
+    m_frag_shader = std::make_unique<wrapper::Shader>(m_device, VK_SHADER_STAGE_FRAGMENT_BIT, "ImGUI fragment shader",
+                                                      "shaders/ui.frag.spv");
+
+    shader_info.stage = VK_SHADER_STAGE_FRAGMENT_BIT;
+    shader_info.module = m_frag_shader->module();
+    shader_info.pName = m_frag_shader->entry_point().c_str();
+
+    shaders.push_back(shader_info);
+
+    // Create font texture
+    unsigned char *fontData;
+    int texWidth, texHeight;
+
+    const std::string font_filename = "assets/fonts/Roboto-Medium.ttf";
```

Comment:
This should be made a `const char *`(or maybe even a `string_view`) to avoid the unnecessary heap allocation.

Reply:
Done.